zivid_description/urdf/macros/zivid_camera.xacro

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://ros.org/wiki/xacro">
+    <xacro:macro name="zivid_camera" params="prefix">
+        <!-- Zivid Properties -->
+        <xacro:property name="M_PI" value="3.1415926535897931" />
+
+        <!-- Zivid Base Link -->
+        <link name="${prefix}base_link">
+            <!-- Visuals -->
+            <visual>
+                <origin xyz="-0.0030 -0.0758 0.0445" rpy="${0.5*M_PI} 0 ${0.5*M_PI}"/>
+                <material name="blue">
+                    <color rgba="0.25 0.25 0.25 1"/>
+                </material>
+                <geometry>
+                    <mesh filename="package://zivid_description/meshes/visual/zivid-one-plus.stl" scale="0.001 0.001 0.001"/>
+                </geometry>
+            </visual>
+
+            <!-- Collisions -->
+            <collision>
+                <origin xyz="-0.0030 -0.0758 0.0445" rpy="${0.5*M_PI} 0 ${0.5*M_PI}"/>
+                <geometry>
+                    <mesh filename="package://zivid_description/meshes/collision/zivid-one-plus.stl" scale="0.001 0.001 0.001"/>
+                </geometry>
+            </collision>
+        </link>
+
+        <!-- Zivid Optical (Measurement) and Projector Frames -->
+        <link name="${prefix}optical_frame"/>
+        <link name="${prefix}projector_frame"/>
+
+        <!-- Zivid Optical (Measurement) and Projector Joints -->
+        <joint name="${prefix}optical_joint" type="fixed">
+            <origin xyz="0.065 0.062 0.0445" rpy="-${0.5*M_PI} 0 -${0.5*M_PI + 8.5/180*M_PI}"/>
+            <parent link="${prefix}base_link"/>
+            <child link="${prefix}optical_frame"/>
+        </joint>
+
+        <joint name="${prefix}projector_joint" type="fixed">
```

Comment:
I would for simplicity not add the projector joint, as I am not sure how much value the fixed projector joint frame will give, as it will mainly give a rough estimate of the actual projector pose.

The reason is that the actual projector will not be at the exact same fixed location for all Zivid One+ cameras, and it will also move very slightly for instance if the camera has experienced a shake or a bump (see addition of experimental in-field correction in SDK 2.2).

@SatjaSivcev can also give his opinion on whether we should keep the projector joint or not.
I suggest removing it. However, I see a possible benefit of keeping it; that even though it's a rough estimate of the projector pose, it could possibly be used to plan the field-of-view and whether the projector will be obstructed by anything in the scene.

Reply:
> However, I see a possible benefit of keeping it; that even though it's a rough estimate of the projector pose, it could possibly be used to plan the field-of-view and whether the projector will be obstructed by anything in the scene.

This was the main reason I put it in indeed.

Reply:
Yeah, I think that is reasonable.
Do you think there is a way we can communicate that it will mainly give a rough estimate of the actual projector pose?

I will double-check the numbers, but IMO it's fine to keep as-is.

Reply:
We can add a README.md just to this package (which then is displayed by GitHub automatically when opening the folder) and/or add comments to the XACRO file where the projector frame is defined

Reply:
Sorry about the back-and-forth, but I discussed with a colleague and I think we prefer to not include the projector frame.
We want to remove the fixed transformation between camera and projector, to avoid any misunderstanding that the transformation between the camera and projector is correct for that specific camera.
In addition, this frame is most valuable if it includes some information of the FOV.

In return I can contribute so that we also get a Zivid URDF/Xacro file for the Zivid Two.

---
I agree with your point that it can be useful to have the transformation between the mounting hole / base link of the camera and a fixed point on the camera which will be the optical/camera frame the point cloud is given in.
As we discuss here
https://github.com/zivid/zivid-ros/pull/17#discussion_r1031358482.

Reply:
I've removed the `projector_frame` and its joint.